pkg/controller/utils/utils.go

```diff
@@ -438,32 +439,33 @@ func WaitToAddResourceWatch(controller controller.Controller, client kubernetes.
 				duration = maxDuration
 			}
 			ticker.Reset(duration)
-			if isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind) {
-				err := controller.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForObject{})
-				if err != nil {
-					log.Info("failed to watch %s resource: %v. Will retry to add watch", obj.GetObjectKind().GroupVersionKind().Kind, err)
-				} else {
-					flag.MarkAsReady()
-					return
-				}
+			if ok, err := isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind); err != nil {
+				log.Info("failed to list registered groups and resources - will retry: %v", err)
```

Comment:
feel like the structure of this is a little bit off... technically this code doesn't know that the returned error means a failure to list registered groups, it could be anything (if for example the implementation of `isResourceReady` changes in the future).

It's pretty nitty because the functions are right next to each other and are unlikely to change, but feels like maybe this log should say "failed to check if resource is ready"

Reply:
This should probably be an `Error` level log as well

Reply:
Eh, maybe it's fine at Info - we'll retry and it's not really breaking. Probably reasonable transient conditions exist where we'll hit this branch and don't want to alarm anyone.